File: etl-common/src/test/scala/org/bigdate/etl/common/scala/test/spark/executors/source/FileSourceExecutor.scala
package org.bigdate.etl.common.scala.test.spark.executors.source


import org.apache.spark.rdd.RDD
import org.apache.spark.sql.SparkSession
import org.bigdata.etl.common.annotations.ETLExecutor
import org.bigdata.etl.common.enums.CommonConstants
import org.bigdata.etl.common.executors.SourceExecutor
import org.bigdate.etl.common.scala.test.spark.configs.FileConfig
import org.slf4j.{Logger, LoggerFactory}

/**
 * Author: GL
 * Date: 2022-04-28
 */
@ETLExecutor("file")
class FileSourceExecutor extends SourceExecutor[SparkSession, RDD[String], FileConfig] {

  val log: Logger = LoggerFactory.getLogger(classOf[FileSourceExecutor])

  override def init(engine: SparkSession, config: FileConfig): Unit = {
    log.info("FileSourceExecutor init, config: {}", config)
  }

  override def process(engine: SparkSession, config: FileConfig): RDD[String] = {
    log.info("FileSourceExecutor process, config: {}", config)
    engine.sparkContext.textFile(config.path, CommonConstants.SECOND)
  }

  override def close(engine: SparkSession, config: FileConfig): Unit = {
    log.info("FileSourceExecutor close, config: {}", config)
  }

  override def check(config: FileConfig): Boolean = {
    log.info("FileSourceExecutor check, config: {}", config.path)
    true
  }

}

File: etl-common/src/test/java/org/bigdata/etl/common/java/test/executors/source/FileSourceExecutor.java
package org.bigdata.etl.common.java.test.executors.source;

import lombok.extern.slf4j.Slf4j;
import org.apache.spark.SparkContext;
import org.apache.spark.rdd.RDD;
import org.bigdata.etl.common.annotations.ETLExecutor;
import org.bigdata.etl.common.java.test.configs.FileConfig;
import org.bigdata.etl.common.executors.SourceExecutor;


/**
 * Author: GL
 * Date: 2022-04-22
 */
@Slf4j
@ETLExecutor("file")
public class FileSourceExecutor implements SourceExecutor<SparkContext, RDD<String>, FileConfig> {

    @Override
    public void init(SparkContext engine, FileConfig config) {
        log.info("FileSourceExecutor init, config: {}", config);
    }

    @Override
    public RDD<String> process(SparkContext engine, FileConfig config) {
        log.info("FileSourceExecutor process, config: {}", config);
        return engine.textFile(config.getPath(), 2);
    }


    @Override
    public void close(SparkContext engine, FileConfig config) {
        log.info("FileSourceExecutor close, config: {}", config);
    }

    @Override
    public boolean check(FileConfig config) {
        log.info("FileSourceExecutor check, config: {}", config.getPath());
        return true;
    }
}
